packages/python-packages/apiview-copilot/src/_database_manager.py

```diff
@@ -93,11 +111,30 @@ def get(self, item_id: str):
 
     def delete(self, item_id: str):
         """
-        Delete an item by its ID. Returns the response from the delete operation.
+        Soft-delete an item by its ID. Sets 'isDeleted' to True instead of removing the document.
         """
         if self.preprocess_id:
             item_id = self.preprocess_id(item_id)
-        return self.client.delete_item(item=item_id, partition_key=item_id)
+        item = self.get(item_id)
+        item["isDeleted"] = True
+        value = self.client.upsert_item(item)
+        self.run_indexer()
+        return value
+
+    def run_indexer(self):
+        """
+        Trigger the Azure Search indexer for this container (examples, guidelines, or memories).
+        """
+        indexer_name = f"{self.container_name}-indexer"
+        search_service = os.environ.get("AZURE_SEARCH_NAME")
+        if not search_service:
+            raise RuntimeError("AZURE_SEARCH_NAME not set in environment.")
+        endpoint = f"https://{search_service}.search.windows.net"
+        client = SearchIndexerClient(endpoint=endpoint, credential=DefaultAzureCredential())
```

Comment:
Would it be possible to use `get_credential()` here from _credential.py instead of DAC?

Reply:
Yep. Updated.